.github/workflows/ci.yaml

```diff
@@ -209,31 +209,46 @@ jobs:
           echo "::set-output name=deployment-name::$DEPLOYMENT_NAME"
           echo "::set-output name=ref-dashed::$REF_DASHED"
 
+          if [ "${{ matrix.environment }}" = "production" ]; then
+            SANDBOX_ID="default"
+          fi
+
+          if [ "${{ matrix.environment }}" = "staging" ]; then
+            SANDBOX_ID="STAGING_SANDBOX_ID"
+          fi
+
+          echo "::set-output name=sandbox-id::$SANDBOX_ID"
+
           yq w -d1 .flux.yaml metadata.name "$DEPLOYMENT_NAME" \
-          | yq w -d1 - metadata.namespace "$DEPLOYMENT_NAME-$REF_DASHED" \
+          | yq w -d1 - metadata.namespace "$DEPLOYMENT_NAME-$SANDBOX_ID" \
+          | yq w -d0 - metadata.name "$DEPLOYMENT_NAME-$SANDBOX_ID" \
           | yq w -d1 - metadata.tags.githubSha "$GITHUB_SHA" \
           | yq w -d1 - spec.releaseName "$DEPLOYMENT_NAME" \
           | yq w -d1 - spec.chart.ref "$GITHUB_REF" \
           | yq w -d1 - spec.values.serviceAccount.iamRole "arn:aws:iam::${{ steps.setup-aws.outputs.aws-account-id }}:role/worker-role-${{ matrix.environment }}" \
-          | yq w -d0 - metadata.name "$DEPLOYMENT_NAME-$REF_DASHED" \
+          | yq w -d1 - spec.values.images.r "$IMAGE_NAME-r" \
+          | yq w -d1 - spec.values.images.python "$IMAGE_NAME-python" \
           > $DEPLOYMENT_NAME.yaml
 
           cat $DEPLOYMENT_NAME.yaml
+        env:
+          IMAGE_NAME: ${{ format('{0}/{1}:{2}', steps.login-ecr.outputs.registry, needs.build-docker.outputs.repo-name, needs.build-docker.outputs.tag) }}
 
+      # TODO: to be deprecated
```

Comment:
why is this going to be deprecated?

Reply:
The chart will have the image names in it as a ConfigFile. Because there can only be one `worker_instance.yaml` file for the entire repo, this is not great, because we can't have a separate one for each staging environment or even staging vs prod.

Now instead of reading from that file, the API will be able to read the image IDs directly from [here](https://github.com/biomage-ltd/iac/blob/master/releases/production/worker.yaml) (under `specs.images.python` and `specs.images.r`) instead of a separate file. This also versions them per deployment which is great.

Reply:
Actually, you can even see it [here](https://github.com/biomage-ltd/iac/blob/master/releases/staging-candidates/worker/refs-pull-33-merge.yaml) now that the pipeline is done.

Reply:
I see, and it will know which folder in `releases` it will need to go so it can read the image id directly